Language/Haskell/GhcMod/ImportedFrom.hs

```diff
@@ -0,0 +1,826 @@
+                                                   setContext $ map (IIDecl . simpleImportDecl . mkModuleName) importList)
+           `gcatch` (\(se :: SomeException)  -> do gmLog GmDebug "qualifiedName" $ strDoc $ "setContext failed with a SomeException, trying to continue anyway..." ++ show se
+                                                   setContext $ map (IIDecl . simpleImportDecl . mkModuleName) importList)
+
+        modSummary <- getModSummary $ mkModuleName targetModuleName :: m ModSummary
+        p <- parseModule modSummary                                 :: m ParsedModule
+        t <- typecheckModule p                                      :: m TypecheckedModule
+
+        let TypecheckedModule{tm_typechecked_source = tcs} = t
+            bs = listifySpans tcs (lineNr, colNr) :: [LHsBind Id]
+            es = listifySpans tcs (lineNr, colNr) :: [LHsExpr Id]
+            ps = listifySpans tcs (lineNr, colNr) :: [LPat Id]
+
+        let bs' = map (showSDocForUser dflags ghcQualify . ppr) bs
+            es' = map (showSDocForUser dflags ghcQualify . ppr) es
+            ps' = map (showSDocForUser dflags ghcQualify . ppr) ps
+
+        gmLog GmDebug "qualifiedName" $ strDoc $ "symbol: " ++ symbol
+        gmLog GmDebug "qualifiedName" $ strDoc $ "line, col: " ++ show (lineNr, colNr)
+
+        let stuff = map dropParens $ concatMap words $ bs' ++ es' ++ ps'
+        gmLog GmDebug "qualifiedName" $ strDoc $ "stuff: " ++ show stuff
+
+        return $ filter (postfixMatch symbol) stuff
+
+  where
+    -- GHC8 starts showing things inside parens? Why? e.g. "(base-4.9.0.0:GHC.Num.+)"
+    dropParens :: String -> String
+    dropParens = dropWhileEnd (== ')') . dropWhile (== '(')
+
+ghcPkgFindModule
+    :: forall m. (GhcMonad m, MonadIO m, GmOut m, GmLog m)
+    => String
+    -> m (Maybe String)
+ghcPkgFindModule mod = do
+    rp <- gmReadProcess
+
+    shortcut [ stackGhcPkgFindModule rp mod
+             , hcPkgFindModule       rp mod
+             , _ghcPkgFindModule     rp mod
+             ]
+  where
+    shortcut :: [m (Maybe a)] -> m (Maybe a)
+    shortcut []     = return Nothing
+    shortcut (a:as) = do
+        a' <- a
+
+        case a' of
+            a''@(Just _)    -> return a''
+            Nothing         -> shortcut as
+
+    optsForGhcPkg :: [String] -> [String]
+    optsForGhcPkg [] = []
+    optsForGhcPkg ("-no-user-package-db":rest)   = "--no-user-package-db"          : optsForGhcPkg rest
+    optsForGhcPkg ("-package-db":pd:rest)        = ("--package-db" ++ "=" ++ pd)   : optsForGhcPkg rest
+    optsForGhcPkg ("-package-conf":pc:rest)      = ("--package-conf" ++ "=" ++ pc) : optsForGhcPkg rest
+    optsForGhcPkg ("-no-user-package-conf":rest) = "--no-user-package-conf"        : optsForGhcPkg rest
+    optsForGhcPkg (_:rest) = optsForGhcPkg rest
+
+    runCmd rp cmd opts = liftIO ((Just <$> (rp cmd opts "")) `catch` (\(_::IOError) -> return Nothing))
```

Comment:
This is no good, you need to get the actual program names/paths from `optPrograms`.